Templates/NewsAndEventsの作成

caramelribon opened the pull request:
## Templates/NewsAndEventsの作成1st
- [x] お手本通りの見た目であるか　ホバー時の挙動、レスポンシブ対応など要確認
- [x] ブランチ名、ファイル名、クラス名、インデントなど、コーディング規則を逸脱していないか
- [x] 記述の重複はないか

リンク
[chromatic](https://62f7a6f53763a1ceb785c6c4-resodxmuab.chromatic.com/)

caramelribon committed:
first commit templates-newsandevents

Hasegawa62 reviewed (commented):
「イベント一覧」の下にもmargin(またはpadding)ほしいです！

また、課題ページにある通り、横並びにしたときにNews と Events の高さが揃っているようにしてください！今まで作成してきたものやfigmaの値と違ってきてしまうかもしれませんが、揃うこと重視でお願いします！

caramelribon committed:
second commit templates-newsandevents

caramelribon commented:
## Templates/NewsAndEventsの作成2nd

イベント一覧の下にmagin-bottomを追加しました。
横並びしたときの左右の高さがそろうように修正しました。

- [x] お手本通りの見た目であるか　ホバー時の挙動、レスポンシブ対応など要確認
- [x] ブランチ名、ファイル名、クラス名、インデントなど、コーディング規則を逸脱していないか
- [x] 記述の重複はないか

リンク
[chromatic](https://62f7a6f53763a1ceb785c6c4-orybzkrigy.chromatic.com/)

minezaki-mutsuki commented on src/components/templates/news-and-events/news-and-events.html:
```diff
@@ -0,0 +1,117 @@
+<div class="news-and-events">
```
大きなまとまりの時にはsectionタグを使うとよいです

minezaki-mutsuki commented on src/components/templates/news-and-events/news-and-events.scss:
```diff
@@ -0,0 +1,19 @@
+@use "../../variables" as *;
+
+.news-and-events {
```
背景色を追加してください

minezaki-mutsuki commented on src/components/templates/news-and-events/news-and-events.scss:
```diff
@@ -0,0 +1,19 @@
+@use "../../variables" as *;
+
+.news-and-events {
+  padding: 56px 0px;
```
この値違うと思います（画面幅狭いときの）

minezaki-mutsuki commented on src/components/templates/news-and-events/news-and-events.scss:
```diff
@@ -0,0 +1,19 @@
+    display: flex;
+    justify-content: space-between;
+    align-items: flex-start;
+    gap: 0px;
```
これなくても大丈夫かと思います

caramelribon committed:
third commit templates-newsandevents

caramelribon commented:
## Templates/NewsAndEventsの作成3rd

.news-and-eventsをdivからsectionに修正しました。
背景をつけました。また、背景が.news-and-eventsの部分すべてにいきわたるよう.containerを加えました。
不要なところを削除しました。
また、paddingについてですが、figmaを見た感じどの画面幅でもpadding 56px 0pxだと思ったんですが、どこを変更したらよろしいでしょうか？

![スクリーンショット (895)](https://user-images.githubusercontent.com/91133956/186628094-4ee30ae8-b2a2-460e-8bce-42cb720e5b01.png)


- [x] お手本通りの見た目であるか　ホバー時の挙動、レスポンシブ対応など要確認
- [x] ブランチ名、ファイル名、クラス名、インデントなど、コーディング規則を逸脱していないか
- [x] 記述の重複はないか

リンク
[chromatic](https://62f7a6f53763a1ceb785c6c4-orybzkrigy.chromatic.com/)

kodai01 commented on src/components/templates/news-and-events/news-and-events.scss:
```diff
@@ -0,0 +1,23 @@
+  margin: 0px;
+
+  .container {
+    padding: 56px 0px;
```
> また、paddingについてですが、figmaを見た感じどの画面幅でもpadding 56px 0pxだと思ったんですが、どこを変更したらよろしいでしょうか？

おそらく、左右の横幅として、右のデザインにもあるとおり16px程度の余白が欲しいと言う意味だと思われます。

kodai01 commented:
それ以外については特に問題はないと思います。

caramelribon committed:
pref: paddingの変更

.conteinarの左右のpaddingの大きさを変更した

その際にbox-sizingをcontent-boxにしないと、paddingaがwidthに含まれてしまうので、content-boxを追加した

forth commit

caramelribon commented:
## Templates/NewsAndEventsの作成4th

.containerのpaddingを修正しました。

- [x] お手本通りの見た目であるか　ホバー時の挙動、レスポンシブ対応など要確認
- [x] ブランチ名、ファイル名、クラス名、インデントなど、コーディング規則を逸脱していないか
- [x] 記述の重複はないか

リンク
[chromatic](https://62f7a6f53763a1ceb785c6c4-orybzkrigy.chromatic.com/)

MoAI522 commented on src/components/organisms/news/news.scss:
```diff
@@ -63,6 +62,10 @@
+
+      &:first-child > a .news-caption {
+        padding: 18.5px 25px 18.5px 19px;
+      }
```
`:not(:first-child)`と`:first-child`は常にどちらか一方が適用されるので、片方を条件なしにしてしまいましょう。
not first childの方が使われる回数が多いので、そちらを`.news-caption`に入れちゃっていいと思います。その代わりにborder-top表示の設定をfirst childにつけましょう。

MoAI522 commented on src/components/templates/news-and-events/news-and-events.scss:
```diff
@@ -0,0 +1,23 @@
+    margin: 0px auto;
+    padding: 56px 16px;
+    @media (min-width: $breakpoint-lg) {
+      max-width: 960px;
```
これをやってしまうと他のセクションのcontainerと幅が合わなくなってしまうのでやめときましょう。これに関してはfigmaと完全に同じじゃなくていいです。

MoAI522 commented on src/components/templates/news-and-events/news-and-events.html:
```diff
@@ -0,0 +1,123 @@
+<section class="news-and-events">
+  <div class="container">
+    <div class="news">
```
bootstrapのgridシステムの使い方をもう一度確認しましょう。
container > row > colの入れ子構造になるべきところが、rowとcolが存在しません。
上手く機能しないので修正お願いします。

MoAI522 commented on src/components/templates/news-and-events/news-and-events.scss:
```diff
@@ -0,0 +1,23 @@
+
+      .events,
+      .news {
+        max-width: 460px;
```
gridシステムを正しく使えればここの指定もいらないはずです。

caramelribon committed:
pref: gridシステムの導入, newsのfirst-childの修正

boostrapのgridシステムを導入しました

organisms/newsの.news-captationにfirst-childをまとめて、:not(:first-child)にpaddingを追加しました

fifth commit

caramelribon commented:
## Templates/NewsAndEventsの作成5th

bootstrapのgridシステムのを導入しました
first-childを.news-captationにまとめて、:not(first-child)のみにしました。

- [x] お手本通りの見た目であるか　ホバー時の挙動、レスポンシブ対応など要確認
- [x] ブランチ名、ファイル名、クラス名、インデントなど、コーディング規則を逸脱していないか
- [x] 記述の重複はないか

リンク
[chromatic](https://62f7a6f53763a1ceb785c6c4-orybzkrigy.chromatic.com/)

MoAI522 reviewed (approved):
LGTM
